### PR TITLE
Single-workflow Signal Command

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

This demonstrates how to write a simple command. Admittedly it didn't have to do much output or output testing and it could have more testing. Also, this does not include the "batch" capabilities of existing CLI.

### CODE REVIEW

File: temporalcli/commands.workflow.go [hunk 1]
```diff
@@ -22,8 +27,33 @@ func (*TemporalWorkflowResetBatchCommand) run(*CommandContext, []string) error {
...
+		Namespace:         c.Parent.Namespace,
+		WorkflowExecution: &common.WorkflowExecution{WorkflowId: c.WorkflowId, RunId: c.RunId},
+		SignalName:        c.Name,
+		Input:             input,
+		Identity:          clientIdentity(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed signalling workflow: %w", err)
+	}
+	cctx.Printer.Println("Signal workflow succeeded")
```
Comment (cretz) [inline temporalcli/commands.workflow.go:55]: Up for discussion on whether this is a log statement or a normal line. If `--output json` is specified, there will be no output regardless.

**Review by dandavison — APPROVED** [review]
Let's merge into the main branch so we can review there.

File: temporalcli/commands.workflow.go (hunk 1 above)
Comment (dandavison) [reply temporalcli/commands.workflow.go:55]: I'm not personally expecting a non-blocking request-response command such as `signal` to have a concept of "logging" -- I'm just expecting stdout and stderr.

Comment (cretz) [reply temporalcli/commands.workflow.go:55]: :+1: This matches existing CLI where, in the absence of more detail, it just prints a line of success.

File: temporalcli/commands.gen.go [hunk 2]
```diff
@@ -631,16 +636,27 @@ func NewTemporalWorkflowShowCommand(cctx *CommandContext, parent *TemporalWorkfl
...
 	s.Parent = parent
 	s.Command.DisableFlagsInUseLine = true
 	s.Command.Use = "signal [flags]"
-	s.Command.Short = "Signal Workflow Execution by Id or List Filter."
-	s.Command.Long = "TODO"
+	s.Command.Short = "Signal Workflow Execution by Id."
+	if hasHighlighting {
+		s.Command.Long = "The \x1b[1mtemporal workflow signal\x1b[0m command is used to Signal a\nWorkflow Execution by ID.\n\n\x1b[1mtemporal workflow signal \\\n\t\t--workflow-id MyWorkflowId \\\n\t\t--name MySignal \\\n\t\t--input '{\"Input\": \"As-JSON\"}'\x1b[0m\n\nUse the options listed below to change the command's behavior."
+	} else {
+		s.Command.Long = "The `temporal workflow signal` command is used to Signal a\nWorkflow Execution by ID.\n\n```\ntemporal workflow signal \\\n\t\t--workflow-id MyWorkflowId \\\n\t\t--name MySignal \\\n\t\t--input '{\"Input\": \"As-JSON\"}'\n```\n\nUse the options listed below to change the command's behavior."
```
Comment (Sushisource) [inline temporalcli/commands.gen.go:653]: Seems odd to have manual color codes, line breaks, etc in these.

**Comment by cretz** [comment]
Closing this in favor of a PR pointing to the cli-rewrite branch (coming soon)

**Pull request closed without merging**